src/states/adult.rs

```diff
@@ -306,19 +306,20 @@ impl Adult {
         }
     }
 
-    // Since we are an adult we will only give info about our section elders and they will further
-    // guide the joining node.
+    // Since we are an adult we should only give info about our section elders and they would
+    // further guide the joining node.
+    // However this lead to a loop if the Adult is the new Elder so we use the same code as
+    // in Elder and return Join in some cases.
```

Comment:
Should we create a github issue for this?

Reply:
https://github.com/maidsafe/routing/issues/1962